File: Data Structures & Algorithms/anagram-groups/submission-2.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,ArrayList<String>> map = new HashMap<>();

        for(int i=0;i<strs.length;i++){

            char[] chars = strs[i].toCharArray(); // Convert to char array
            Arrays.sort(chars);                   // Sort the array in-place
            String key = new String(chars); 
            if (!map.containsKey(key)) {
                map.put(key, new ArrayList<>()); // Create if it doesn't exist
            }
            map.get(key).add(strs[i]);
        }

        return new ArrayList<>(map.values());
    }
}

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        hash_map = {}

        for i in s:
            if i in hash_map:
                hash_map[i]+=1 
            else:
                hash_map[i] = 1


        for i in t:
            if i in hash_map:
                hash_map[i]-=1 
            else:
                False
        
        for value in hash_map.values():
            if value != 0:
                return False

        return True


        